### PR TITLE
[RESTEASY-2581] Add profile to enable testing using Wildfly "standalone-microprofile.xml" configuration file

### PR DESCRIPTION
**Pull request by fabiobrz**
Fixes https://issues.redhat.com/browse/RESTEASY-2581

Still WIP because some test cases still need for a little investigation.
Basically, this PR introduces a new profile which - like for what happens in Wildfly TS - would use the `standalone-microprofile.xml` config file to start the server up.
Then, when running tests by passing a new property - namely `-Dts.standalone.microprofile` to activate the above mentioned profile, some tests in `integration-tests` and `integration-tests-spring/inmodule` TS modules will fail.
Hence, two new JUint `Category` interfaces have been added:

* `ExpectedFailingBecauseOfSmallRyeMicroprofileOpenApi11`: skips tests that _seem_ to fail because of the SmallRye MP OpenAPI implementation used in Wildfly 19 (further investigation to confirm this is needed)
* `ExpectedFailingWithStandaloneMicroprofileConfiguration`: skips tests that fail because of missing subsystems in Wildfly when `standalone-microprofile.xml` is used.

This PR also bumps  `server.version` to 19.0.0.Final.

Commands to verify that TS is running fine with the new profile:
```
$ export SERVER_VERSION=19.0.0.Final
$ mvn -B -am -pl integration-tests -fae -Dserver.version=$SERVER_VERSION install -Dts.standalone.microprofile
```

### CODE REVIEW

**Comment by rsearls** [comment]
Shouldn't information about -Dts.standalone.microprofile be added to the README.MD?

File: testsuite/integration-tests/pom.xml [hunk 1]
```diff
@@ -181,6 +181,28 @@
...
+                </property>
+            </activation>
+            <properties>
+                <jboss.server.config.file.name>standalone-microprofile.xml</jboss.server.config.file.name>
+                <!--
+                    Excluding those tests that would need modules 
+                    which are in standalone-full.xml but not in 
+                    standalone-microprofile.xml
+                -->
+                <additional.surefire.excluded.groups>org.jboss.resteasy.category.ExpectedFailingWithStandaloneMicroprofileConfiguration,org.jboss.resteasy.category.ExpectedFailingBecauseOfSmallRyeMicroprofileOpenApi11</additional.surefire.excluded.groups>
```
Comment (marekkopecky) [inline testsuite/integration-tests/pom.xml:203]: ExpectedFailingBecauseOfSmallRyeMicroprofileOpenApi11 issue needs to be investigated.

**Comment by marekkopecky** [comment]
> Shouldn't information about -Dts.standalone.microprofile be added to the README.MD?

+1 (cc @tommaso-borgato )

**Comment by fabiobrz** [comment]
Hi @rsearls and @marekkopecky, I added details to the README.md. Tests are passing with JDK11 and JDK8 locally, so I removed the WIP.

**Comment by marekkopecky** [comment]
@fabian I don't prefer to merge this now, because we don't have the reasons of OpenAPI errors (AFAIK). Thank you for readme updates.

**Comment by fabiobrz** [comment]
I closed and reopened the PR in order to trigger Travis CI again as it failed despite the build succeed both with JDK8 and JDK11 (OpenJDK) locally.

**Comment by ronsigal** [comment]
@fabiobrz, I merged the PR for branch 3.12 and then saw @marekkopecky's remark here. What do you think about the OpenAPI errors? If you think they need more investigation, could you create a JIRA for investigating

**Comment by fabiobrz** [comment]
Hi @ronsigal, I see that a PR to produce a release to solve the issues with MP OpenAPI has been filed already: https://github.com/smallrye/smallrye-open-api/pull/325 and I think the release will come soon and hope that the artifact version will be bumped in Wildfly soon, as well.
@marekkopecky WDYT?

**Comment by marekkopecky** [comment]
`If you think they need more investigation, could you create a JIRA for investigating`
Investigation is done by  @tommaso-borgato ( https://issues.redhat.com/browse/JBEAP-19495 ). In my PoV, this PR should contain links or additional description of OpenAPI annotation. But if the issue will be fixed soon, we will have to remove this OpenAPI annotation at all soon as well :)

**Comment by marekkopecky** [comment]
`In my PoV, this PR should contain links or additional description of ExpectedFailingBecauseOfSmallRyeMicroprofileOpenApi11 annotation`

Thinking about this, this can be done in some following MR, so I'm ok with merging this MR now

**Comment by marekkopecky** [comment]
@ronsigal (cc @tommaso-borgato  ) Please let me know, if you prefer to add jira links to this PR directly or if you want any other changes in this PR

**Comment by ronsigal** [comment]
Hi @marekkopecky,

Thanks for looking at this. Actually, I merged https://github.com/resteasy/Resteasy/pull/2407 into branch 3.12 yesterday. Branch 3.11 is dormant, except, maybe, for backporting something.

I want to release 3.12.0.Final today, to leave a little time before the feature freeze. If https://github.com/smallrye/smallrye-open-api/pull/325 were merged today, that would be good, but, since we're talking about tests, I don't think it's that urgent. I also suggested a new JIRA for pursuing the failures for 3.13.0.Final.

-Ron

**Comment by fabiobrz** [comment]
Hi @marekkopecky (CC @ronsigal and @tommaso-borgato), this is just to clarify that `ExpectedFailingBecauseOfSmallRyeMicroprofileOpenApi11` is about failure due to SmallRye OpenAPI 1.1 implementation - as detailed in the Javadoc comment to the interface - rater than to the JDK(11).

**Comment by fabiobrz** [comment]
Hi @ronsigal and CC @marekkopecky, RESTEasy JIRA was filed to track issues with MP OpenAPI, see https://issues.redhat.com/browse/RESTEASY-2589

**Comment by ronsigal** [comment]
I'll close this since it's superceded by https://github.com/resteasy/Resteasy/pull/2407, now merged into branch 3.12.

**Comment by marekkopecky** [comment]
> Actually, I merged #2407 into branch 3.12 yesterday. Branch 3.11 is dormant, except, maybe, for backporting something.

@ronsigal actually, this backporting would  be useful for QE, I added also some details to [RESTEASY-2589](https://issues.redhat.com/browse/RESTEASY-2589) jira. Can you please consider to merge this PR?

**Comment by ronsigal** [comment]
Hi @marekkopecky,

Ok, sure, I don't think it should be a problem. I didn't know 3.11 was of any interest.

-Ron

**Comment by marekkopecky** [comment]
@ronsigal Thanks, let me know if you want some additional actions from us (you know, it's still not merged).

**Comment by ronsigal** [comment]
Hi @marekkopecky,

Sorry, I forgot about this one. By the way, I see that https://github.com/smallrye/smallrye-open-api/pull/325 has been closed. Are you able to upgrade smallrye-openapi?

-Ron

**Comment by ronsigal** [comment]
@marekkopecky, I'll wait for the tests to run and then merge.

**Comment by fabiobrz** [comment]
Thanks @ronsigal! FYI and also CC @marekkopecky and @tommaso-borgato I noticed that some tests in the `spring-inmodule` module might have actually failing due to the SmallRye OpenAPI < 1.1.22 issue. I'll verify this later on but in such case an adjustment could be needed for this PR as well as a new one for the 3.12 branch. Of course - as all of the relevant tests here involve Wildfly integration - we must wait to have the SmallRye version bump in there or at least bump it by ourselves and see how it goes. 
@pferraro maybe you know when the component version will be upgraded in Wildfly? 
Hope this makes sense and let me know what you think.

**Comment by pferraro** [comment]
@fabiobrz WildFly uses version 1.2.3 of smallrye-open-api.

**Comment by ronsigal** [comment]
@marekkopecky, sorry for the delay.

**Comment by fabiobrz** [comment]
Thanks @pferraro! I actually didn’t check the master branch. I see that these tests use Wildfly 19. My concern was abut this. I’ll take a closer look :)

**Comment by ronsigal** [comment]
Hey @marekkopecky, @fabiobrz, @tommaso-borgato,

It looks like the changes in https://github.com/smallrye/smallrye-open-api/pull/325 have been applied to smallrye-open-api 1.2.3.

RESTEasy 3.11.2.Final doesn't have any dependency on smallrye-open-api, direct or indirect, so I'm guessing all that RESTEasy would need would be removing the `ExpectedFailingBecauseOfSmallRyeMicroprofileOpenApi11` tags, right? Would you guys need a 3.11.3.FInal release

**Comment by fabiobrz** [comment]
Hi @ronsigal, the tests we’re talking about are supposed to verify integration with Wildfly (those in the spring-inmodule and spring-deployment as well). If I don’t go wrong (I can’t verify right now) 3.12 and 3.11 are using Wildfly 19.0.0.Beta/Final and those are producing the failures for which the annotation is needed at the moment. If the Wildfly version those branch would use is going to change to one which supports the new SmallRye OpenAPI release, then yes, I’d say the “ ExpectedFailingBecauseOfSmallRyeMicroprofileOpenApi11“ annotation could be removed, IMO.

**Comment by ronsigal** [comment]
Hey @fabiobrz, we have a default WF version in the code, and then we have the versions we test with in travis / github actions. Right now, 3.x versions are testing with WF 18 and 19, and we should add 20 to that.

But I was under the impression that the ts.standalone.microprofile profile would be activated in WildFly or EAP testing. If that's true, then it would be enough to remove `@ExpectedFailingBecauseOfSmallRyeMicroprofileOpenApi11` from any RESTEasy version that's targeting a version of WF / EAP that's got the upgraded smallrye-open-api, right? So for example, I suppose 3.13.0.Final will go into WF 21, so the annotation should be removed there, and there's no rush on that. But what about this PR into RESTEasy 3.11.2.Final? My understanding is that it will go into EAP 7.3.1 and EAP XP 1.0.0. Will they have the upgraded smallrye-open-api? If so, then you might want a RESTEasy 3.11.3.Final, right?

**Comment by fabiobrz** [comment]
Hi @ronsigal, I think you summarised the whole thing perfectly. I am on the same page but I’d like to hear from @marekkopecky As well, mostly on what’s going on with EAP.
Cheers!

**Comment by fabiobrz** [comment]
> Thanks @ronsigal! FYI and also CC @marekkopecky and @tommaso-borgato I noticed that some tests in the `spring-inmodule` module could be actually failing due to the SmallRye OpenAPI < 1.1.22 issue. I'll verify this later on but in such case an adjustment could be needed for this PR as well as a new one for the 3.12 branch. 
...

Hey @ronsigal, meantime I've checked this today and it seems to me that SmallRye OpenAPI implementation is unrelated with respect to tests in `integration-tests/spring/inmodule` so I've left `ExpectedFailingWithStandaloneMicroprofileConfiguration` on those to be skipped. Basically no need for another PR, neither here nor for `3.12`.

**Comment by fabiobrz** [comment]
Hi again @ronsigal. 

> But I was under the impression that the ts.standalone.microprofile profile would be activated in WildFly or EAP testing. If that's true, then it would be enough to remove @ExpectedFailingBecauseOfSmallRyeMicroprofileOpenApi11 from any RESTEasy version that's targeting a version of WF / EAP that's got the upgraded smallrye-open-api, right? 

Yes, I do agree.

> So for example, I suppose 3.13.0.Final will go into WF 21, so the annotation should be removed there, and there's no rush on that. 

Yep, agree on this too and - a s Paul stated earlier - it seems that the annotation could be removed from 3.12 too, which seems to be used by Wildfly 20 (https://github.com/wildfly/wildfly/blob/master/pom.xml#L391) which in turn is equipped with SmallRye OpenAPI 1.2.3, https://github.com/wildfly/wildfly/blob/master/pom.xml#L305 (fixed, I guess).

> But what about this PR into RESTEasy 3.11.2.Final? My understanding is that it will go into EAP 7.3.1 and EAP XP 1.0.0. Will they have the upgraded smallrye-open-api?

Those have the upgraded SmallRyeOpenAPI indeed. I've been checking the POM on which both the above mentioned distributions are based, and it uses RESTEasy 3.11.2 and SmallRye OpeAPI 1.1.22 (fixed).

> If so, then you might want a RESTEasy 3.11.3.Final, right?

Well, yes. My understanding is that if you release a 3.11.3.Final _withouth_ the `ExpectedFailingBecauseOfSmallRyeMicroprofileOpenApi11` annotations, then next releases of 7.3.x and XP streams could buomp the RESTEasy version to that one.

So, basically, IIUC, we need to file 2 PRs to remove the annotation both from 3.11 and 3.12 as it seems no longer needed now. I can file them. WDYT? CC @marekkopecky, @tommaso-borgato

**Comment by marekkopecky** [comment]
> we need to file 2 PRs to remove the annotation both from 3.11 and 3.12 as it seems no longer needed now

+1
